Core/include/Acts/Propagator/MultiEigenStepperLoop.ipp

```diff
@@ -43,37 +43,9 @@ auto MultiEigenStepperLoop<E, R, A>::boundState(
     return MultiStepperError::AllComponentsConversionToBoundFailed;
   }
 
-  // To ensure we are in perigee representation when taking the mean over the
-  // components in bound representation, we do the mean over the free
-  // represenation and then go back to bound representation.
-  Acts::FreeVector freeMean = Acts::FreeVector::Zero();
-  for (const auto& [w, p, c] : states) {
-    freeMean += w * Acts::detail::transformBoundToFreeParameters(
-                        surface, state.geoContext, p);
-  }
-  
-  std::cout << "mean before intersect: " << freeMean.transpose() << std::endl;
-
-  freeMean.segment<3>(eFreePos0) =
-      surface
-          .intersect(state.geoContext, freeMean.segment<3>(eFreePos0),
-                     freeMean.segment<3>(eFreeDir0), false)
-          .template intersection.position;
-          
-  std::cout << "mean after intersect: " << freeMean.transpose() << std::endl;
-
-  // TODO Needed to reduce tolerance here in order to avoid free-to-bound-errors
-  auto boundMeanRes = Acts::detail::transformFreeToBoundParameters(
-      freeMean, surface, state.geoContext);
-  if(not boundMeanRes.ok()) {
-    std::cout << boundMeanRes.error().message() << std::endl;
-  }
-  assert(boundMeanRes.ok());
-
-  // TODO Not sure if this is entirely correct, visit again later
-  const auto cov =
+  const auto [mean, cov] =
       detail::angleDescriptionSwitch(surface, [&](const auto& desc) {
-        return detail::combineCov(states, *boundMeanRes, 1.0, Acts::Identity{}, desc);
+        return detail::combineGaussianMixture(states, Acts::Identity{}, desc);
```

Comment:
I see it was there before, but where does `Acts::Identity` actually come from??

Reply:
@benjaminhuth

Reply:
hmm from `Utilities/Identity.hpp`. Basically from C++20 on it is 
```c++
namespace Acts {
using Identity = std::identity;
}
```
Before, it is a backport of this. I use this projection-pattern a bit in the GSF code, inspired from the ranges library.

Reply:
I was unaware of this.